pkg/controller/ibmblockcsi/ibmblockcsi_controller.go

```diff
@@ -338,16 +339,31 @@ func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI, o
 		return err
 	}
 
-	instance.Status.ControllerReady = controller.Status.ReadyReplicas == controller.Status.Replicas
+	instance.Status.ControllerReady = controllerStatefulset.Status.ReadyReplicas == controllerStatefulset.Status.Replicas
 	instance.Status.NodeReady = node.Status.DesiredNumberScheduled == node.Status.NumberAvailable
 	phase := csiv1.DriverPhaseNone
 	if instance.Status.ControllerReady && instance.Status.NodeReady {
 		phase = csiv1.DriverPhaseRunning
 	} else {
-		if originalStatus.ControllerReady && !instance.Status.ControllerReady {
-			controllerRestart = true
+		if !instance.Status.ControllerReady {
+			err := r.getControllerPod(controllerStatefulset, controllerPod)
```

Comment:
what is this for?

Reply:
it fills the `controllerPod` instance with (non-empty) values

Reply:
Don't we want to do that if the controller is ready as well?

Reply:
don't see why